File: LibraryApp/Application/Publishers/PublisherService.cs
using Application.Publishers.DTOs;
using AutoMapper;
using Domain.Publishers;

namespace Application.Publishers;

public class PublisherService : IPublisherService
{
    private readonly IMapper _mapper;
    private readonly IPublisherRepository _publisherRepository;
    
    public PublisherService(IMapper mapper, IPublisherRepository publisherRepository)
    {
        _mapper = mapper;
        _publisherRepository = publisherRepository;
    }

    public List<PublisherDto> GetAll()
    {
        var publishers = _publisherRepository.GetAll();
        return _mapper.Map<List<PublisherDto>>(publishers);
    }

    public PublisherDto GetById(int id)
    {
        var publisher = _publisherRepository.GetById(id);
        return _mapper.Map<PublisherDto>(publisher);
    }

    public void Insert(PublisherDto publisher)
    {
        var publisherEntity = _mapper.Map<Publisher>(publisher);
        publisherEntity.IsActive = true;
        _publisherRepository.Insert(publisherEntity);
    }

    public void Update(PublisherDto publisher)
    {
        var publisherEntity = _mapper.Map<Publisher>(publisher);
        publisherEntity.IsActive = true;
        _publisherRepository.Update(publisherEntity);
    }

    public void Delete(int id)
    {
        _publisherRepository.Delete(id);
    }
}

File: angular/src/app/features/publishers/service/publisher.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PublisherCreateDto, PublisherDto } from '../models/publisher.dto';
import { environment } from '../../../../environments/environment.development';

@Injectable({
    providedIn: 'root'
})
export class PublisherService {
    private api_url = `${environment.apiUrl}/api/publishers`;

    constructor(private http: HttpClient){}

    getAll(): Observable<PublisherDto[]>{
        return this.http.get<PublisherDto[]>(this.api_url);
    }

    getById(id: number): Observable<PublisherDto>{
        return this.http.get<PublisherDto>(`${this.api_url}/${id}`);
    }

    Insert(publisher: PublisherCreateDto): Observable<PublisherDto>{
        return this.http.post<PublisherDto>(this.api_url, publisher);
    }

    Update(id: number, publisher: PublisherDto): Observable<void>{
        return this.http.put<void>(`${this.api_url}/${id}`, publisher);
    }

    Delete(id: number): Observable<void>{
        return this.http.delete<void>(`${this.api_url}/${id}`);
    }
}
